src/cargo/ops/lockfile.rs

```diff
@@ -68,10 +74,29 @@ pub fn write_lockfile(dst: &Path, resolve: &Resolve) -> CargoResult<()> {
         None => {}
     }
 
+    // Load the original lockfile if it exists.
+    if let Ok(orig) = load_string(dst) {
+        if has_crlf_line_endings(&orig) {
+            out = out.replace("\n", "\r\n");
+        }
+        if out == orig {
```

Comment:
Could you add a comment here saying that if nothing changed we avoid rewriting, and this helps us work on things like readonly filesystems?